From 83714071c68c588c66972fc0fe0f5f2a57a7dd55 Mon Sep 17 00:00:00 2001
From: sudhanshukumar22 <sudhanshu.kumar@broadcom.com>
Date: Thu, 21 Nov 2024 20:57:11 -0800
Subject: [PATCH] When the file is config replayed, we cannot handle the errors
  in command execution. We would let all commands in the file to complete, 
 irrespective of some of them failing. Also, certain protocols like RIP, RIPng
 are not supported in SONIC. So, it is expected that their configuration files
 will not be present.

---
 lib/vty.c               | 16 +++++++++++++---
 mgmtd/mgmt_be_adapter.c |  2 +-
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/lib/vty.c b/lib/vty.c
index 1c9cff478..017e0c7c4 100644
--- a/lib/vty.c
+++ b/lib/vty.c
@@ -2654,7 +2654,9 @@ void vty_read_file_finish(struct vty *vty, struct nb_config *config)
 		nl = strchr(ve->error_buf, '\n');
 		if (nl)
 			*nl = '\0';
-		flog_err(EC_LIB_VTY, "%s on config line %u: %s", message,
+		/* When the file is config replayed, we cannot handle the errors in command execution.
+			We would let all commands in the file to complete, irrespective of some of them failing. */
+		flog_warn(EC_LIB_VTY, "%s on config line %u: %s", message,
 			 ve->line_num, ve->error_buf);
 	}
 
@@ -2773,7 +2775,11 @@ FILE *vty_open_config(const char *config_file, char *config_default_dir)
 				flog_warn(EC_LIB_BACKUP_CONFIG,
 					  "using backup configuration file!");
 			else {
-				flog_err(
+				/* There are few protocols e.g. RIP, RIPng, which are ununsed in SONIC.
+				   So, there will not be any config file for them.
+				   Flagging error, will be considered a breakage by loganalyzer, whereas
+				   it is valid for these protocol config file to be not present. */
+				flog_warn(
 					EC_LIB_VTY,
 					"%s: can't open configuration file [%s]",
 					__func__, config_file);
@@ -2822,7 +2828,11 @@ FILE *vty_open_config(const char *config_file, char *config_default_dir)
 					  "using backup configuration file!");
 				fullpath = config_default_dir;
 			} else {
-				flog_err(EC_LIB_VTY,
+				/* There are few protocols e.g. RIP, RIPng, which are ununsed in SONIC.
+				   So, there will not be any config file for them.
+				   Flagging error, will be considered a breakage by loganalyzer, whereas
+				   it is valid for these protocol config file to be not present. */
+				flog_warn(EC_LIB_VTY,
 					 "can't open configuration file [%s]",
 					 config_default_dir);
 				goto tmp_free_and_out;
diff --git a/mgmtd/mgmt_be_adapter.c b/mgmtd/mgmt_be_adapter.c
index b311bf469..58f866115 100644
--- a/mgmtd/mgmt_be_adapter.c
+++ b/mgmtd/mgmt_be_adapter.c
@@ -732,7 +732,7 @@ static void mgmt_be_adapter_conn_init(struct event *thread)
 	 * transaction in progress.
 	 */
 	if (mgmt_txn_notify_be_adapter_conn(adapter, true) != 0) {
-		zlog_err("XXX txn in progress, retry init");
+		zlog_warn("XXX txn in progress, retry init");
 		mgmt_be_adapter_sched_init_event(adapter);
 		return;
 	}
-- 
2.39.4

